src/app/src/components/Geocoder.js

```diff
@@ -0,0 +1,156 @@
+import { usePreventMapDoubleClick } from '../hooks';
+
+const GEOCODER_WIDTH = 320;
+
+const decodeJson = response => response.json();
+
+export default function Geocoder({ isOpen, onClose }) {
+    const dispatch = useDispatch();
+    const ref = usePreventMapDoubleClick();
+    const [inputValue, setInputValue] = useState('');
+    const [suggestions, setSuggestions] = useState([]);
+    const [highlightIndex, setHighlightIndex] = useState(-1);
+    const [loading, setLoading] = useState(false);
+
+    useEffect(() => {
+        if (isOpen && ref.current) {
+            ref.current.focus();
+        }
+    }, [isOpen, ref]);
+
+    const updateSuggestions = search => {
+        if (search.length > 0) {
+            fetch(getGeocodeSuggestUrl(search))
+                .then(decodeJson)
+                .then(({ suggestions }) => setSuggestions(suggestions));
+        } else {
+            setSuggestions([]);
+        }
+    };
+
+    const preventTabBlur = event => {
+        if (event.key === 'Tab') {
+            event.preventDefault();
+        }
+    };
+
+    const nothingHighlighted = highlightIndex === -1;
+
+    const keyUpHandler = ({ key }) => {
+        switch (key) {
+            case 'Down': // IE/Edge specific value
+            case 'ArrowDown':
+            case 'Tab':
+                moveHighlightDown();
+                break;
+
+            case 'Up': // IE/Edge specific value
+            case 'ArrowUp':
+                moveHighlightUp();
+                break;
+
+            case 'Enter':
+                doGeocode(
+                    nothingHighlighted
+                        ? { text: inputValue }
+                        : suggestions[highlightIndex]
+                );
+                break;
+
+            case 'Esc': // IE/Edge specific value
+            case 'Escape':
+                onClose();
+                break;
+
+            default:
+                return;
+        }
+    };
+
+    const moveHighlightDown = () => {
+        setHighlightIndex(index => (isLastIndex(index) ? 0 : index + 1));
+    };
+
+    const isLastIndex = index => index === suggestions.length - 1;
+
+    const moveHighlightUp = () => {
+        setHighlightIndex(index => Math.max(0, index - 1));
+    };
+
+    const doGeocode = suggestion => {
+        setInputValue(suggestion.text);
+        setSuggestions([]);
+        setLoading(true);
+
+        fetch(getGeocodeUrl(suggestion))
+            .then(decodeJson)
+            .then(({ candidates }) => {
+                setLoading(false);
+                dispatch(setGeocodeResult(candidates));
+            });
+    };
+
+    const clearResult = () => dispatch(clearGeocodeResult());
+
+    const hasSuggestions = suggestions.length > 0;
+
+    return (
+        <ScaleFade in={isOpen}>
+            <Input
+                position='relative'
+                ref={ref}
+                bg='white'
+                focusBorderColor='gray.900'
```

Comment:
@rajadain @mstone121 I just updated the 'Input field (focus state)' component to have a 'grey-500' outline instead of pure black. That should make it lest stark.

https://www.figma.com/file/tJPZ0lkT5C3SAyS2g2R1aA/IoW---Boundary-Sync-Tool?node-id=234%3A12160

Reply:
Thanks. Let's change this to

```suggestion
                focusBorderColor='gray.500'
```

Reply:
de46bc8f6f33baba09a4db3afb15701febe03a87